runtime-infra/pn-log-analytics.yaml

```diff
@@ -0,0 +1,529 @@
+                      {"name": "s3_event_time", "type": "timestamp"},
+                      {"name": "s3_event_name", "type": "varchar"},
+                      {"name": "s3_user_identity", "type": "varchar"},
+                      {"name": "s3_source_ip", "type": "varchar"},
+                      {"name": "s3_bucket_name", "type": "varchar"},
+                      {"name": "s3_object_key", "type": "varchar"},
+                      {"name": "s3_object_size", "type": "bigint"},
+                      {"name": "s3_object_etag", "type": "varchar"},
+                      {"name": "s3_object_version_id", "type": "varchar"}
+                    ]
+                  }
+                - DatabaseName: !Ref GlueDatabase
+                  TableName: !Ref GlueTable
+
+  GlueCrawlerRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - glue.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole
+      Policies:
+        - PolicyName: S3Access
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:GetBucketLocation
+                  - s3:ListBucket
+                  - s3:GetObject
+                Resource:
+                  - !Sub arn:aws:s3:::${LogsBucketName}
+                  - !Sub arn:aws:s3:::${LogsBucketName}/*
+        - PolicyName: S3EncryptionAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:GetEncryptionConfiguration
+                Resource:
+                  - !Sub arn:aws:s3:::${LogsBucketName}
+        - PolicyName: KMSEncryptionAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - kms:Decrypt
+                  - kms:DescribeKey
+                Resource:
+                  - !Ref LogsBucketKmsKeyArn
+
+  GlueCrawler:
+    Type: AWS::Glue::Crawler
+    Properties:
+      Name: LogAnalyticsCrawler
+      Role: !GetAtt GlueCrawlerRole.Arn
+      DatabaseName: !Ref GlueDatabase
+      Targets:
+        CatalogTargets:
+          - DatabaseName: !Ref GlueDatabase
+            Tables:
+              - !Ref GlueTable
+      SchemaChangePolicy:
+        UpdateBehavior: LOG
+        DeleteBehavior: LOG
+      Configuration: |
+        {
+          "Version": 1.0,
+          "CrawlerOutput": {
+            "Partitions": {
+              "AddOrUpdateBehavior": "InheritFromTable"
+            }
+          }
+        }
+      TablePrefix: ''
+      # Schedule:
```

Comment:
@alessioangelotti-pagopa se non specifichi l'espressione cron, ogni quanto gira il crawler ?

Reply:
@fdammassa per il crawler ho scelto di lasciarlo disabilitato in questa fase, per una questione di riduzione costi( seppur bassi circa 30 cents al giorno).
 In questo modo è impostato su "on-demand", l'intenzione è quella di sostituire il crawler con un altro meccanismo di aggiornamento partizioni (e.g lambda) solo per questa tabella relativa ai log cloudwatch.

Reply:
@fdammassa ok aggiunta condition e parametro per valore cron expression.